dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/test/webapi/H2ControllerIntegrationTestBase.java

```diff
@@ -28,13 +28,16 @@
 package org.hisp.dhis.test.webapi;
 
 import org.hisp.dhis.test.IntegrationH2Test;
+import org.hisp.dhis.test.config.H2TestConfig;
 import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
 
 /**
  * Base class for all Spring Mock MVC based controller tests which use an H2 in-memory DB.
  *
  * @author Jan Bernitt
  */
-@ActiveProfiles("test-h2")
 @IntegrationH2Test
+@ActiveProfiles("test-h2")
```

Comment:
Is this profile still needed?

Reply:
Yes, we still have production code that is aware of profiles like https://github.com/dhis2/dhis2-core/blob/6971c9463dd3b4f49dce24c50ade9bf316b0b814/dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/database/HibernateDatabaseInfoProvider.java#L164. We might be able to solve this particular case. I have not tried.

I tried to get rid of prod code knowing about test profiles in general in https://github.com/dhis2/dhis2-core/pull/18282 but am blocked. The issue is that some tests need to set specific `dhis.conf` properties. We do this via `@ContextConfiguration` overriding beans. We use `@Conditional` like https://github.com/dhis2/dhis2-core/blob/db09c459bcbe7dcbf1be4581dc8c6122bac6efc8/dhis-2/dhis-support/dhis-support-cache-invalidation/src/main/java/org/hisp/dhis/cacheinvalidation/redis/CacheInvalidationListener.java#L64 to create certain components only if a feature is active. So there is a chicken and egg situation 😅 Spring Boot has automatic support for what we would like to do with `application.properties` and [ConditionalOnProperty](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/autoconfigure/condition/ConditionalOnProperty.html). It would be amazing if we can come up with something similar in https://dhis2.atlassian.net/browse/DHIS2-17797. @jbee let's talk about this when you are ready to look into https://dhis2.atlassian.net/browse/DHIS2-17797.